src/common/input/handler.rs

```diff
@@ -276,34 +282,160 @@ impl InputHandler {
     }
 }
 
+lazy_static! {
+    static ref KEY_ORDER: HashMap<Key, i32> = {
+        let mut m = HashMap::new();
+        m.insert(Key::Left, 0);
+        m.insert(Key::Right, 0);
+        m.insert(Key::Up, 1);
+        m.insert(Key::Down, 1);
+        m.insert(Key::PageUp, 2);
+        m.insert(Key::PageDown, 2);
+        m
+    };
+}
+
+/// Get a prior key from keybinds
+/// many keys may be mapped to one action, e.g. kj/↑↓
+/// but we do not want to show all of them in help info,
+/// so just pickup one primary key.
+fn get_major_key_by_action(keybinds: &ModeKeybinds, action: &[Action]) -> Key {
+    let mut key = Key::Null;
+    for (k, actions) in &keybinds.0 {
+        if actions == action {
+            if key == Key::Null {
+                // old key is null
+                key = *k;
+            } else if let Some(new_order) = KEY_ORDER.get(k) {
+                if let Some(old_order) = KEY_ORDER.get(&key) {
+                    if new_order < old_order {
+                        // old key has lower order (larger number) than new one
+                        key = *k;
+                    }
+                } else {
+                    // old key does not have order, new key have order
+                    // then use new keybind
+                    key = *k;
+                }
+            }
+        }
+    }
+    key
+}
+
+fn get_key_map_string(key_config: &ModeKeybinds, actions: &[&[Action]]) -> String {
+    let map = actions
+        .iter()
+        .map(|&actions| get_major_key_by_action(&key_config, actions))
+        .map(|key| key.to_string())
+        .collect::<Vec<_>>();
+    let should_split = map.iter().any(|s| s.chars().count() > 1);
+    map.into_iter().fold(String::new(), |s0, s| {
+        if !s0.is_empty() && should_split {
+            format!("{}/{}", s0, s)
+        } else {
+            format!("{}{}", s0, s)
+        }
+    })
+}
+
 /// Creates a [`Help`] struct indicating the current [`InputMode`] and its keybinds
 /// (as pairs of [`String`]s).
 // TODO this should probably be automatically generated in some way
-pub fn get_mode_info(mode: InputMode) -> ModeInfo {
+pub fn get_mode_info(mode: InputMode, key_config: &Keybinds) -> ModeInfo {
+    let key_config = key_config
+        .0
+        .get(&mode)
+        .cloned()
+        .unwrap_or_else(|| Keybinds::get_defaults_for_mode(&mode));
     let mut keybinds: Vec<(String, String)> = vec![];
```

Comment:
A part of me has wanted for a while to change this to a `HashMap<Key, Action>` and to shift all of the formatting to the status-bar plugin, since that's the only place this is actually used. This would likely be a little bit of extra work, but I'd certainly support the effort!

Reply:
Now `<Key, Vec<Action>>` will be passed to plugin and mode_info is generated in `status-bar`